apps/remix-ide/src/app/panels/file-panel.js

```
@@ -96,6 +97,20 @@ module.exports = class Filepanel extends ViewPlugin {
     return this.el
   }
 
+  /**
+   *
+   * @param item { name: string, type?: string[], path?: string[], extension?: string[], pattern?: string[] }
+   * @param callback (...args) => void
+   */
+  registerContextMenuItem (item, callback) {
+    if (!item.name || !callback) return console.error('menu name and callback is mandatory')
+    if (!item.type && !item.path && !item.extension && !item.pattern) return console.error('invalid file matching criteria provided')
```

you could also throw an error here, so it bounces back to the caller